modules/api/php/endpoints/candidate/visit/instrument/flags.class.inc

```diff
@@ -127,6 +127,120 @@ class Flags extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PUT request
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request) : ResponseInterface
+    {
+        // TODO :: Check permissions. How??
+        $this->_instrumentStatus = new \NDB_BVL_InstrumentStatus();
+        $this->_instrumentStatus->select($this->_instrument->commentID);
+
+        $data = json_decode((string) $request->getBody(), true);
+
+        $requiredfields = [
+            'Candidate',
+            'Visit',
+            'DDE',
+            'Instrument',
+        ];
+
+        $diff = array_diff($requiredfields, array_keys($data['Meta']));
+        if (!empty($diff)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Field(s) missing in Meta: ' . implode(', ', $diff)
+            );
+        }
+
```

Comment:
it is missing the verification that the required fields values correspond to $this->_visit and $this->_instrument values.

Reply:
Somehow, I thought the `Meta` fields were required, but they are not, so I am just removing them and it won't be necessary to verify them (unless I misunderstood your request)

Reply:
They are required to match `$this->_visit->getCandID()` , `$this->_visit->getVisitLabel()`, `$isDDE = strpos($instrumentdata['CommentID'], 'DDE_') === 0;` and `$instrumentname = $this->_instrument->testName`